### PR TITLE
GShootList: Reflect search query in URL

### PR DESCRIPTION
**Pull request by petersutter**
**What this PR does / why we need it**:
This PR updates the `GShootList` component to reflect the search query in the URL. When the `shootSearch` changes, the URL is updated so users can bookmark the URL with the search query. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user
Reflect search query on project clusters page in URL
```

### CODE REVIEW

**Comment by grolu** [comment]
I see this warning in the logs:

Screenshot 2024-07-05 at 11 43 34

File: frontend/src/views/GShootList.vue [hunk 1]
```diff
@@ -311,15 +315,39 @@ export default {
...
+
+    function setSearchFromUrl () {
+      const searchQuery = route.query.search || ''
+      shootSearch.value = searchQuery
+      debouncedShootSearch.value = searchQuery
+    }
+
+    onMounted(() => {
+      setSearchFromUrl()
+    })
```
Comment (holgerkoser) [inline frontend/src/views/GShootList.vue:340]: I would use `vueuse` composable in this case
```js
    const params = useUrlSearchParams('hash-params')
    const shootSearch = ref(params.q)
    const debouncedShootSearch = ref(shootSearch.value)

    watch(params, value => {
      if (value.q !== shootSearch.value) {
        debouncedShootSearch.value = shootSearch.value = value.q
      }
    })

    watch(debouncedShootSearch, value => {
      params.q = value
    })
```

Comment (petersutter) [reply frontend/src/views/GShootList.vue]: ✅

**Comment by petersutter** [comment]
The warning is now resolved, thanks to @holgerkoser's suggestion

File: frontend/src/views/GShootList.vue [hunk 2]
```diff
@@ -311,15 +313,29 @@ export default {
...
+    watch(params, value => {
+      if (value.q !== shootSearch.value) {
+        debouncedShootSearch.value = shootSearch.value = value.q
+      }
+    })
+
+    watch(shootSearch, value => {
+      params.q = value
+    })
+
```
Comment (holgerkoser) [inline frontend/src/views/GShootList.vue:329]: * If I change the hash value in the URL the `searchQuery` is not updated correctly due to the `resetSearchQuery` call in the navigation guards.

* To delete the parameter q in the URL we should set the value of `params.q` to `null`. I would not delete the property of the reactive object.
  
I would propose to change the code like this:
```js
    const params = useUrlSearchParams('hash-params')
    const shootSearch = ref(params.q)
    const debouncedShootSearch = ref(shootSearch.value)

    function setShootSearch (value) {
      shootSearch.value = value
      setDebouncedShootSearch()
    }

    function resetShootSearch () {
      debouncedShootSearch.value = shootSearch.value = params.q
    }

    const setDebouncedShootSearch = debounce(() => {
      debouncedShootSearch.value = shootSearch.value
    }, 500)

    watch(() => params.q, value => {
      if (shootSearch.value !== value) {
        setShootSearch(value)
      }
    })

    watch(shootSearch, value => {
      if (!value) {
        params.q = null
      } else if (params.q !== value) {
        params.q = value
      }
    })
```

Comment (petersutter) [reply frontend/src/views/GShootList.vue]: ✅

**Review by holgerkoser — APPROVED** [review]
/lgtm

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**